src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -71,6 +71,38 @@ protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, S
                                               declarationSyntax.Span.IntersectsWith(candidateLocation.SourceSpan)));
         }
 
+        /// <summary>
+        /// If container is a tuple type, any of its tuple element which has a friendly name will cause
+        /// the suppression of the corresponding default name (ItemN).
+        /// In that case, Rest is also removed.
+        /// </summary>
+        protected static IEnumerable<ISymbol> SuppressDefaultTupleElements(INamespaceOrTypeSymbol container,
+            IEnumerable<ISymbol> symbols)
+        {
+            if (!container.IsType)
+            {
+                return symbols;
+            }
+
+            var type = (ITypeSymbol)container;
+            if (!type.IsTupleType)
+            {
+                return symbols;
+            }
+
+            var tuple = (INamedTypeSymbol)type;
```

Comment:
@jcouv just pointed that !s.IsImplicitlyDeclared would exclude synthesized Item1 unconditionally, but we want to include it if that is the only name we have for element1. 

So, without TupleElementIndex being public (or some other API), filtering by name is the easiest option for now. :-(